crates/sui-graphql-rpc/src/types/type_filter.rs

```diff
@@ -88,6 +100,40 @@ impl TypeFilter {
     }
 }
 
+impl FqNameFilter {
+    /// Modify `query` to apply this filter, treating `package` as the column containing the package
+    /// address, `module` as the module containing the module name, and `name` as the column
+    /// containing the module member name.
+    pub(crate) fn apply<P, M, N, QS, ST, GB>(
```

Comment:
This filter bundles processing the graphql input to querying against the database schema. If tx_calls' schema changes for some reason, say package remains a separate field but module and name gets bundled into a single column, we would need to introduce a new filter or modify the existing one to instead take in package: P and name: MN

In comparison, something like paginate_query defers the actual pagination logic to the type using the method, so changes to the db schema is localized to that type.

I wonder if it would make sense for FqNameFilter, TypeFilter, etc. to return the processed tuple of values for (p,m,n) or (p,m,t) and have the type implement the querying logic. Or something like, apply(...p: Fn(string) -> query) so that in tx blocks, it'd be called as .apply(...p: Fn(string) -> tx_calls.package.filter(string))

Reply:
> I wonder if it would make sense for FqNameFilter, TypeFilter, etc. to return the processed tuple of values for (p,m,n) or (p,m,t) and have the type implement the querying logic. Or something like, apply(...p: Fn(string) -> query) so that in tx blocks, it'd be called as .apply(...p: Fn(string) -> tx_calls.package.filter(string))

That's something worth thinking about at the point where we actually have two forms to abstract over.